js/popover.js

```diff
@@ -64,9 +64,22 @@
   Popover.prototype.getContent = function () {
     var $e = this.$element
     var o  = this.options
+    var contentType = typeof o.content
+
+    if (!/string|function/.test(contentType)) {
+      throw new Error('Popover received type ${' + contentType + '}. Was expecting type string or a function.')
+    }
+
+    if (contentType == 'function') {
+      var contentReturnType = typeof o.content.call($e[0])
+      if (!/string/.test(contentReturnType) && !(o.content.call($e[0]) instanceof $)) {
```

Comment:
Also<sup>2</sup>, this doesn't handle DOM nodes only jQuery object. Plus it's less heavy if you check for a jQuery object by using `returnValueOfContentFunction.jquery`, see http://api.jquery.com/jquery-2/.